src/FirmwarePlugin/FirmwarePlugin.cc

```diff
@@ -50,6 +50,11 @@ FirmwarePluginFactoryRegister* FirmwarePluginFactoryRegister::instance(void)
     return _instance;
 }
 
+FirmwarePlugin::FirmwarePlugin(MAV_TYPE vehicleType)
```

Comment:
You can't do this. There is a single FirmwarePlugin created for all PX4 vehicles. The same one just keeps getting handed out.

Reply:
Ha! So all those calls (such as `supportsNegativeThrust()`) must pass a `Vehicle` instance then as there is no other way to figure it out.